src/pages/demoProductItem.js

```diff
@@ -29,35 +30,48 @@ class DemoProductItem extends Component {
 
   addItemToCart = () => {
     let currentCartItems = localStorage.getItem('demo-cart') ? JSON.parse(localStorage.getItem('demo-cart')) : {};
-    if(currentCartItems[this.props.product.id]) {
-      currentCartItems[this.props.product.id].purchaseQuantity += this.state.itemCount
+    if (currentCartItems[this.props.pathContext.productId]) {
```

Comment:
`this.props.pathContext.productId` can be taken out in a variable and use that variable wherever needed. reduce repeated code.